api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -589,17 +603,17 @@ private static String sanitizeNumber(Number value, String type) {
     return "(" + numDigits + "-digit-" + type + ")";
   }
 
-  private static String sanitizeString(CharSequence value, long now, int today) {
+  private static String sanitizeString(CharSequence value, long nowMillis, int today) {
     try {
       if (DATE.matcher(value).matches()) {
         Literal<Integer> date = Literal.of(value).to(Types.DateType.get());
         return sanitizeDate(date.value(), today);
       } else if (TIMESTAMP.matcher(value).matches()) {
-        Literal<Long> ts = Literal.of(value).to(Types.TimestampType.withoutZone());
-        return sanitizeTimestamp(ts.value(), now);
+        Literal<Long> ts = Literal.of(value).to(Types.TimestampType.nanosWithoutZone());
```

Comment:
Do we always want to interpret it as NANOs? Depending on the datatype that's being used this might restrict the range of dates.

Reply:
Resolved in 1a6cf52c1baa35e1fdd086d8907fbed7873cf317